internal/service/pushbasedlogexport/resource.go

```diff
@@ -0,0 +1,174 @@
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newTFModel)...)
+}
+
+func (r *pushBasedLogExportRS) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var tfState TFPushBasedLogExportRSModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &tfState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := r.Client.AtlasV2
+	projectID := tfState.ProjectID.ValueString()
+	logConfig, _, err := connV2.PushBasedLogExportApi.GetPushBasedLogConfiguration(ctx, projectID).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError("error getting push-based log export configuration", err.Error())
+		return
+	}
+
+	newTFModel, diags := NewTFPushBasedLogExport(ctx, projectID, logConfig, &tfState.Timeouts)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newTFModel)...)
+}
+
+func (r *pushBasedLogExportRS) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var tfPlan TFPushBasedLogExportRSModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &tfPlan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	logExportConfigReq := NewPushBasedLogExportReq(&tfPlan)
+
+	connV2 := r.Client.AtlasV2
+	projectID := tfPlan.ProjectID.ValueString()
+	if _, err := connV2.PushBasedLogExportApi.UpdatePushBasedLogConfiguration(ctx, projectID, logExportConfigReq).Execute(); err != nil {
+		resp.Diagnostics.AddError("error during push-based log export configuration update", err.Error())
+		return
+	}
+
+	timeout, diags := tfPlan.Timeouts.Update(ctx, defaultTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	logExportConfigResp, err := WaitStateTransition(ctx, projectID, connV2.PushBasedLogExportApi,
+		retryTimeConfig(timeout, minTimeoutCreateUpdate))
+	if err != nil {
+		resp.Diagnostics.AddError("error during push-based log export configuration update", err.Error())
+		return
+	}
+
+	newTFModel, diags := NewTFPushBasedLogExport(ctx, projectID, logExportConfigResp, &tfPlan.Timeouts)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newTFModel)...)
+}
+
+func (r *pushBasedLogExportRS) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var tfState *TFPushBasedLogExportRSModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &tfState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := r.Client.AtlasV2
+	projectID := tfState.ProjectID.ValueString()
+	if _, err := connV2.PushBasedLogExportApi.DeletePushBasedLogConfiguration(ctx, projectID).Execute(); err != nil {
+		resp.Diagnostics.AddError("error during push-based log export configuration delete", err.Error())
+		return
+	}
+
+	deleteTimeout, diags := tfState.Timeouts.Delete(ctx, defaultTimeout)
```

Comment:
qq: what is this Timeouts when it's created or deleted?

Reply:
I've kept the default timeout to 15min.
If the user doesn't specify a timeout block in their config explicitly, the default timeout value will be used.